src/components/Datatable/Datatable.tsx

```diff
@@ -83,8 +89,12 @@ const Datatable = <D extends Record<string, unknown>>({
       }}
     >
       <StyledDatatable flexDirection="column">
-        <FlexContainer>ToolbarModule</FlexContainer>
-        <FlexContainer>Filters</FlexContainer>
+        {controlsConfig.isControlsEnabled && (
+          <ControlModule
+            {...defaultControlsConfig}
```

Comment:
There is merged config on line 69 to 72, you should be using that. This will use just the default properties and it cannot be affected from outside.

Reply:
updated to use custom controls config

Reply:
Now it uses just the config passed as a property. There is this line
```js
const {
    defaultHiddenColumns,
    defaultColumnOrder,
  }: ControlsConfig<D> = mergeRight(defaultControlsConfig, controlsConfig);
```
sou you should add `...restControlsConfig` in the destructure and then do 
```js
<ControlModule
  defaultHiddenColumns={defaultHiddenColumns}
  defaultColumnOrder={defaultColumnOrder}
  {...restControlsConfig}
/>